ui/components/Policies/PolicyList/PolicyTable.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react";
+
+import moment from "moment";
+import { FC } from "react";
+
+import { useFeatureFlags } from "../../../hooks/featureflags";
+import { formatURL } from "../../../lib/nav";
+import { V2Routes } from "../../../lib/types";
+import DataTable, { filterConfig } from "../../DataTable";
+import Link from "../../Link";
+import Text from "../../Text";
+import PolicyMode from "../Utilis/PolicyMode";
+import Severity from "../Utilis/Severity";
+import { Policy } from "../../../lib/api/core/core.pb";
+
+interface CustomPolicy extends Policy {
+  audit?: string;
+  enforce?: string;
+}
+
+interface Props {
+  policies: CustomPolicy[];
+}
+
+export const PolicyTable: FC<Props> = ({ policies }) => {
+  const { isFlagEnabled } = useFeatureFlags();
+
+  policies.forEach((policy) => {
+    policy.audit = policy.modes?.includes("audit") ? "audit" : "";
+    policy.enforce = policy.modes?.includes("admission") ? "enforce" : "";
+  });
+
+  let initialFilterState = {
+    ...filterConfig(policies, "severity"),
+    ...filterConfig(policies, "enforce"),
+    ...filterConfig(policies, "audit"),
+  };
+
+  if (
+    isFlagEnabled("WEAVE_GITOPS_FEATURE_TENANCY") &&
+    isFlagEnabled("WEAVE_GITOPS_FEATURE_CLUSTER")
+  ) {
+    initialFilterState = {
+      ...initialFilterState,
+      ...filterConfig(policies, "tenant"),
+    };
+  }
+
+  if (isFlagEnabled("WEAVE_GITOPS_FEATURE_CLUSTER")) {
+    initialFilterState = {
+      ...initialFilterState,
+      ...filterConfig(policies, "clusterName"),
+    };
+  }
+  return (
+    <DataTable
+      key={policies?.length}
+      filters={initialFilterState}
+      rows={policies}
+      fields={[
+        {
+          label: "Policy Name",
+          value: ({ clusterName, id, name }) => (
+            <Link
+              to={formatURL(V2Routes.PolicyDetailsPage, {
+                clusterName: isFlagEnabled("WEAVE_GITOPS_FEATURE_CLUSTER")
```

Comment:
you can bind `clusterName` to the parameter value in line 63, right ?

Reply:
i update the `ClusterName` value in the same function that apply `audit` and `enforce` values